<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jacques van Helden" />


<title>Command-line use of bioinformatics tools</title>

<script src="blast_proteome_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="blast_proteome_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="blast_proteome_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="blast_proteome_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="blast_proteome_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="blast_proteome_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="blast_proteome_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="blast_proteome_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="blast_proteome_files/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #303030; color: #cccccc; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; }
td.sourceCode { padding-left: 5px; }
pre, code { color: #cccccc; background-color: #303030; }
code > span.kw { color: #f0dfaf; } /* Keyword */
code > span.dt { color: #dfdfbf; } /* DataType */
code > span.dv { color: #dcdccc; } /* DecVal */
code > span.bn { color: #dca3a3; } /* BaseN */
code > span.fl { color: #c0bed1; } /* Float */
code > span.ch { color: #dca3a3; } /* Char */
code > span.st { color: #cc9393; } /* String */
code > span.co { color: #7f9f7f; } /* Comment */
code > span.ot { color: #efef8f; } /* Other */
code > span.al { color: #ffcfaf; } /* Alert */
code > span.fu { color: #efef8f; } /* Function */
code > span.er { color: #c3bf9f; } /* Error */
code > span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */
code > span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code > span.sc { color: #dca3a3; } /* SpecialChar */
code > span.vs { color: #cc9393; } /* VerbatimString */
code > span.ss { color: #cc9393; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #f0dfaf; } /* ControlFlow */
code > span.op { color: #f0efd0; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code > span.at { } /* Attribute */
code > span.do { color: #7f9f7f; } /* Documentation */
code > span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code > span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code > span.in { color: #7f9f7f; font-weight: bold; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Command-line use of bioinformatics tools</h1>
<h4 class="author"><em>Jacques van Helden</em></h4>
<h4 class="date"><em>Created: 8/10/2018; last update: 2018-10-09</em></h4>

</div>


<hr />
<div id="prerequisite" class="section level1">
<h1>Prerequisite</h1>
<p>This practical assumes that you are connected to a cluster of the IFB NNCR. This enables you to benefit from a software environment with pre-installed software tools for the different fields of bioinformatics.</p>
<p>To open a connection to the IFB NNCR, follow the totirla <a href="../NNCR-cluster_connection/NNCR-cluster_connection.html">Tutorial</a> |</p>
</div>
<div id="resources" class="section level1">
<h1>Resources</h1>
<table style="width:96%;">
<colgroup>
<col width="33%" />
<col width="62%" />
</colgroup>
<thead>
<tr class="header">
<th>Resource</th>
<th>URL and description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Ensembl Bacteria</td>
<td><a href="https://bacteria.ensembl.org/index.html" class="uri">https://bacteria.ensembl.org/index.html</a> <br>Database and browser for bacterial and archaeal genomes maintained at the European Bioinformatics Institute (EBI)</td>
</tr>
<tr class="even">
<td>GTF format</td>
<td><a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format4" class="uri">https://genome.ucsc.edu/FAQ/FAQformat.html#format4</a> <br>Specification of the Gene Transfer Format (GTF) on the UCSC genome browser Web site</td>
</tr>
</tbody>
</table>
</div>
<div id="tutorial" class="section level1">
<h1>Tutorial</h1>
<div id="browsing-genomes-with-ensemblgenomes" class="section level2">
<h2>Browsing genomes with EnsemblGenomes</h2>
<p>As reference genome, we will use the strain <em>K12 MG1655</em> of <em>Escherichia coli</em>.</p>
<ol style="list-style-type: decimal">
<li><p>In a web browser, open a connexion to Ensembl Bacteria (<a href="https://bacteria.ensembl.org" class="uri">https://bacteria.ensembl.org</a>).</p></li>
<li><p>In the <strong>Search for a genome</strong> box, type <em>MG1655</em>. The interface will display a list of the genomes matching this string. Select the strain with taxonomic ID (TAXID) <em>511145</em>.</p></li>
</ol>
<div class="figure" style="text-align: center">
<img src="images/ensembl_bacteria_organism_selection.png" alt="**EnsemblGenomes Bacteria organism choice**. " width="60%" />
<p class="caption">
<strong>EnsemblGenomes Bacteria organism choice</strong>.
</p>
</div>
<ol start="3" style="list-style-type: decimal">
<li>After having selected your bacteria of interest, type <em>aspartokinase</em> in the search box and start the search.</li>
</ol>
<div class="figure" style="text-align: center">
<img src="images/ensembl_bacteria_search_aspartokinase.png" alt="**EnsemblGenomes gene search**. " width="60%" />
<p class="caption">
<strong>EnsemblGenomes gene search</strong>.
</p>
</div>
<ol start="4" style="list-style-type: decimal">
<li>Read the annotation of the three first matches. Note the names of the genes coding for these three enzymes.</li>
</ol>
<div class="figure" style="text-align: center">
<img src="images/Ecoli_aspartokinase_search_result.png" alt="**EnsemblGenomes search result for *aspartokinase***. " width="60%" />
<p class="caption">
<strong>EnsemblGenomes search result for <em>aspartokinase</em></strong>.
</p>
</div>
</div>
<div id="exercise" class="section level2">
<h2>Exercise</h2>
<p>We will use blast on the command line to search for homologs of <em>E.coli</em> aspartokinases in other bacterial genomes.</p>
<p>Each student will chose a different bacteria to perform the search.</p>
</div>
<div id="downloading-protein-sequences-and-gene-annotations-from-ensembl-genome" class="section level2">
<h2>Downloading protein sequences and gene annotations from Ensembl genome</h2>
<ol style="list-style-type: decimal">
<li><p>In a separate window, open a connection to <a href="https://bacteria.ensembl.org/">EnsemblGenomes Bacteria</a> (<a href="https://bacteria.ensembl.org/" class="uri">https://bacteria.ensembl.org/</a>).</p></li>
<li><p>Click on the **Downloads* link (you can also access it directly here: <a href="https://bacteria.ensembl.org/info/website/ftp/" class="uri">https://bacteria.ensembl.org/info/website/ftp/</a>).</p></li>
<li><p>In the <strong>filter</strong> box, type MG1655. This select the subset of genomes whose name contains this string.</p></li>
</ol>
<div class="figure" style="text-align: center">
<img src="images/ensemblgenomes_bacteria_ftp_mg1655.png" alt="**EnsemblGenomes download filter**. " width="60%" />
<p class="caption">
<strong>EnsemblGenomes download filter</strong>.
</p>
</div>
<ol start="4" style="list-style-type: decimal">
<li><p>Click on the <strong>FASTA</strong> link in the column <strong>Protein sequences</strong>/</p></li>
<li><p>and select <strong>Copy link address</strong>.</p></li>
</ol>
<div class="figure" style="text-align: center">
<img src="images/protein_fasta_link.png" alt="**EnsemblGenomes** Link to download all protein sequences for a selected organism. " width="200" />
<p class="caption">
<strong>EnsemblGenomes</strong> Link to download all protein sequences for a selected organism.
</p>
</div>
<ol start="5" style="list-style-type: decimal">
<li><p>Come back to the Unix terminal of the IFB NNCR.</p></li>
<li><p>Create a directory for the current practical.</p></li>
</ol>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">mkdir</span> -p ~/bacteria_blast
<span class="bu">cd</span> ~/bacteria_blast</code></pre></div>
<ol start="7" style="list-style-type: decimal">
<li>type <code>wget -rNL</code> and paste the content of the clipboard. This should give the following command (note: at the tiem you run this tutorial, the release may be different, this one is from October 2018).</li>
</ol>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">wget</span> --no-clobber ftp://ftp.ensemblgenomes.org/pub/release-41/bacteria//fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655/pep/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.pep.all.fa.gz

<span class="co">## Check that the file is there</span>
<span class="fu">ls</span> -l

<span class="co">## Uncompress the file</span>
<span class="fu">gunzip</span> Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.pep.all.fa.gz

<span class="co"># Measure the size of the uncompressed file</span>
<span class="fu">du</span> -k Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.pep.all.fa

<span class="co"># Check the 20 fist lines of the file</span>
<span class="fu">head</span> -n 20 Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.pep.all.fa

<span class="co"># Check the 20 last lines of the file</span>
<span class="fu">tail</span> -n 20 Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.pep.all.fa</code></pre></div>
</div>
<div id="downloading-genome-annotations" class="section level2">
<h2>Downloading genome annotations</h2>
<p>After having downloaded the protein sequences, we will download gene annotations. In EnsemblGenomes, annotations are available in different formats, including GTF, a text file with columns separated by tabulations.</p>
<ol style="list-style-type: decimal">
<li><p>We In the browser, come back to the EnsemblGenomes Bacteria Download site, select MG1655 and follow the same instructions as above to download and uncompress the GTF file.</p></li>
<li><p>Uncompress the gzi archive.</p></li>
</ol>
</div>
<div id="exercise-genome-annotations" class="section level2">
<h2>Exercise: genome annotations</h2>
<p>As an exercise, you will now elaborate Unix commands to explore the content of E.coli genome annotations.</p>
<p>For this, we recommmend to use and combine the following commands. In order to fine-tune the parameters, read their manual on the Unix terminal by typing <code>man [command_name]</code> (for example <code>man wc</code>).</p>
<table style="width:94%;">
<colgroup>
<col width="13%" />
<col width="80%" />
</colgroup>
<thead>
<tr class="header">
<th>Command</th>
<th>Usage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>wc</td>
<td>count lines, words and characters of a file</td>
</tr>
<tr class="even">
<td>cut</td>
<td>cut specific columns of a text file</td>
</tr>
<tr class="odd">
<td>grep</td>
<td>select rows of a file that match a given query (string, regular expression)</td>
</tr>
<tr class="even">
<td>awk</td>
<td>a powerful tool to work on text files, combining row filters, column selections, and many other features.</td>
</tr>
<tr class="odd">
<td>sort</td>
<td>sort the rows of a text file based on user-specified columns. The sorting can be done either alphabetically or numerically, in ascending or descending order</td>
</tr>
<tr class="even">
<td>uniq</td>
<td>discard redundant lines from a text file. The option <code>-c</code> counts the number of occurrences of each line in a redundant file</td>
</tr>
</tbody>
</table>
<p>You can learn more Unix commands by typing “Linux cheat sheet” in any Web search engine.</p>
<p>We ask you to write all the answers in a text file, with comments, in order to enable other people to reproduce all your commands.</p>
<div id="questions" class="section level3">
<h3>Questions</h3>
<ol style="list-style-type: decimal">
<li>How many rows contains the GTF file for E.coli K12 MG1655 ?</li>
<li><p>How many annotation rows are there in the GTF file (for this, you need to discard the comment rows starting with a <code>#</code> character) ?</p></li>
<li><p>How many of these rows correspond to genes ?</p></li>
</ol>
</div>
</div>
<div id="activating-the-conda-environment-for-blast-on-the-nncr-cluster" class="section level2">
<h2>Activating the conda environment for blast on the NNCR cluster</h2>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="bu">source</span> <span class="va">$CONDA_HOME</span>/bin/activate blast-2.7.1</code></pre></div>
<hr />
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
